packages/shell-api/src/collection.ts

```diff
@@ -2388,17 +2388,30 @@ export default class Collection extends ShellApiWithMongoClass {
   @apiVersions([])
   // TODO(MONGOSH-1471): use SearchIndexDescription once available
   async createSearchIndex(
-    indexName?: string | Document,
+    indexNameOrOptions?: string | Document,
```

Comment:
could we define a stricter type here instead of `Document`?

Reply:
Well, we can make it `string | SearchIndexDescription  | Document`, where:

```
interface SearchIndexDescription {
  definition: Document;
  name?: string;
  type?: string;
}
```

But `definition` will still have the `Document` type. Not sure if this brings a lot of value here.

Reply:
I think it's still good to have even as purely documentation

Reply:
I can add it if this helps to read the code 👍

Reply:
Yes, I think that's a great idea – and maybe it's worth adding [TS function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads) to this method just to describe what the actual valid call signatures are 🙂